internal/util/kafka.go

```diff
@@ -65,6 +63,83 @@ func (k kafkautil) NewWriter(topic string) *kafka.Writer {
 	}
 }
 
+func getSaslAndTLSConfig() (sasl.Mechanism, *tls.Config) {
+	username := config.DefaultConfig.KafkaUsername
+	password := config.DefaultConfig.KafkaPassword
+	saslmechanismName := config.DefaultConfig.KafkaSaslMechanism
+
+	tlsConfig, err := createTLSConfig(config.DefaultConfig.KafkaCAPath)
+	if err != nil {
+		fmt.Println("Error creating TLS configuration for Kafka:", err)
+		tlsConfig = &tls.Config{} // Providing default empty TLS configuration
+	}
+
+	saslMechanism, err := createSaslMechanism(
+		saslmechanismName,
+		username,
+		password,
+	)
+	if err != nil {
+		fmt.Println("Error creating SASL Mechanism for Kafka, using plain mechanism:", err)
+	}
+
+	return saslMechanism, tlsConfig
+}
+
+func createTLSConfig(pathToCert string) (*tls.Config, error) {
+
+	tlsConfig := tls.Config{
+		MinVersion: tls.VersionTLS12,
+	}
+
+	if pathToCert == "" {
+		return &tlsConfig, nil
+	}
+
+	caCert, err := ioutil.ReadFile(pathToCert)
```

Comment:
ioutil is deprecated as of Go 1.16 I think. You should use `os.ReadFile` instead.

Reply:
Thanks for pointing that out 👍🏻. Fixed it!